addons/point_of_sale/models/pos_session.py

```diff
@@ -1571,11 +1573,61 @@ def _load_model(self, model):
             raise NotImplementedError(_("The function to load %s has not been implemented.", model))
 
     def load_pos_data(self):
-        loaded_models = {}
-        self = self.with_context(loaded_models=loaded_models)
+        loaded_data = {}
+        self = self.with_context(loaded_data=loaded_data)
         for model in self._pos_ui_models_to_load():
-            loaded_models[model] = self._load_model(model)
-        return loaded_models
+            loaded_data[model] = self._load_model(model)
+        self._pos_data_process(loaded_data)
+        return loaded_data
+
+    def _get_attributes_by_ptal_id(self):
+        product_attributes = self.env['product.attribute'].search([('create_variant', '=', 'no_variant')])
+        product_attributes_by_id = {product_attribute.id: product_attribute for product_attribute in product_attributes}
+        domain = [('attribute_id', 'in', product_attributes.mapped('id'))]
+        product_template_attribute_values = self.env['product.template.attribute.value'].search(domain)
+        key = lambda ptav: (ptav.attribute_line_id.id, ptav.attribute_id.id)
+        res = {}
+        for key, group in groupby(sorted(product_template_attribute_values, key=key), key=key):
+            attribute_line_id, attribute_id = key
+            values = [{**ptav.product_attribute_value_id.read(['name', 'is_custom', 'html_color'])[0],
+                       'price_extra': ptav.price_extra} for ptav in list(group)]
+            res[attribute_line_id] = {
+                'id': attribute_line_id,
+                'name': product_attributes_by_id[attribute_id].name,
+                'display_type': product_attributes_by_id[attribute_id].display_type,
+                'values': values
+            }
+
+        return res
+
+    def _pos_data_process(self, loaded_data):
+        """
+        This is where we need to process the data if we can't do it in the loader/getter
+        """
+        loaded_data['version'] = exp_version()
+
+        loaded_data['units_by_id'] = {unit['id']: unit for unit in loaded_data['uom.uom']}
+
+        loaded_data['taxes_by_id'] = {tax['id']: tax for tax in loaded_data['account.tax']}
+        tax_ids = loaded_data['taxes_by_id'].keys()
+        for real_tax in self.env['account.tax'].browse(tax_ids).get_real_tax_amount():
```

Comment:
you can create a field "real_tax_amount" in accounting module, which will be computed and stored, you'll get this value through the search_read.

This function and call to this had been made in stable version as a fix when we could not create a field